packages/refresh-theme/config/omeda-identity-x.js

```diff
@@ -1,114 +1,110 @@
+  } = args;
+  return {
+    clientKey: omedaConfig.clientKey,
+    brandKey: omedaConfig.brandKey,
+    appId: omedaConfig.appId,
+    inputId: omedaConfig.inputId,
+    rapidIdentProductId,
+    idxConfig,
 
-  /**
-   * Behavior config is now mandatory and can be generated by the CLI.
-   * @see https://github.com/parameter1/identity-x-omeda-cli
-   *  */
-  behaviors: {
-    logIn: 6270,
-    verifyEmail: 6269,
-    submitProfile: 6271,
-  },
-  behaviorAttributes: {
-    website: {
-      id: 127,
-      valueId: websiteBehaviorAttributeId,
+    /**
+     * Behavior config is now mandatory and can be generated by the CLI.
+     * @see https://github.com/parameter1/identity-x-omeda-cli
+     *  */
+    behaviors: {
+      logIn: 6270,
+      verifyEmail: 6269,
+      submitProfile: 6271,
     },
-    actionSource: {
-      id: 125,
-      valueIds: {
-        default: 451143,
-        newsletterSignup: 451144,
-        comments: 451145,
-        contentGate: 451142,
+    behaviorAttributes: {
+      website: {
+        id: 127,
+        valueId: websiteBehaviorAttributeId,
       },
-    },
-    newsletterSignupType: {
-      id: 126,
-      valueIds: {
-        default: 451148,
-        pushdown: 451147,
-        inlineContent: 451150,
-        inlineSection: 451149,
-        footer: 451146,
+      actionSource: {
+        id: 125,
+        valueIds: {
+          default: 451143,
+          newsletterSignup: 451144,
+          comments: 451145,
+          contentGate: 451142,
+        },
       },
-    },
-    contentGateType: {
-      id: 128,
-      valueIds: {
-        default: 451151,
-        metered: 451152,
-        printPreview: 451153,
+      newsletterSignupType: {
+        id: 126,
+        valueIds: {
+          default: 451148,
+          pushdown: 451147,
+          inlineContent: 451150,
+          inlineSection: 451149,
+          footer: 451146,
+        },
+      },
+      contentGateType: {
+        id: 128,
+        valueIds: {
+          default: 451151,
+          metered: 451152,
+          printPreview: 451153,
+        },
       },
     },
-  },
-  appendPromoCodeToHook,
-  appendBehaviorToHook,
-  appendDemographicToHook,
-  onAuthenticationSuccessFormatter: (async ({ req, payload }) => {
-    // BAIL if omedaGraphQLCLient isnt available return payload.
-    if (!req.$omedaGraphQLClient) return payload;
-
-    const identityXOptInHooks = req.app.locals.site.getAsObject('identityXOptInHooks');
-    const omeda = req.app.locals.site.getAsObject('omeda');
-    if (identityXOptInHooks.onAuthenticationSuccess) {
-      const { autoSignupDeploymentTypes } = identityXOptInHooks.onAuthenticationSuccess;
+    appendPromoCodeToHook,
+    appendBehaviorToHook,
+    appendDemographicToHook,
+    onAuthenticationSuccessFormatter: (async ({ req, payload }) => {
+      // BAIL if omedaGraphQLCLient isnt available return payload.
+      if (!omedaConfig.omedaGraphQLClientProp || !onAuthenticationSuccess) return payload;
```

Comment:
The `omedaGraphQLClientProp` config has a default value and ~is~ _should be_ required when configuring the Omeda package. If the value isn't specified, this will properly return, but if an incorrect value is specified (such as `ogqlv2` when no middleware sets it) or the Omeda package isn't correctly configured, this won't be correct.

What you actually want to check that the GraphQL client can be found based on the config prop path, so you should check that the value returned from the request object isn't falsey:

```suggestion
      const omedaGraphQLClient = req[omedaConfig.omedaGraphQLClientProp];
      if (!omedaGraphQLClient || !onAuthenticationSuccess) return payload;
```